Improve gpg subprocess timeout and stop using custom process module

lukpueh opened the pull request:
### Description of the changes being introduced by the pull request:

gpg interface function to sign and export public keys now have an optional subprocess timeout. This means, users no longer need to patch a global in order to configure this value. Moreover, the default timeout is now defined inside the gpg subpackage, and increased to 10.
--> The should reduce test failures on slow runners, and makes it more intuitive to override the default.

Internally the gpg subpackage now no longer uses the custom subprocess wrapper, which provides very little benefit over calling directly into stdlib `subprocess`. 
--> This is in preparation for deprecation of the process module

see commit message for details

### Please verify and check that the pull request fulfils the following requirements:

- [ ] The code follows the [Code Style Guidelines](https://github.com/secure-systems-lab/code-style-guidelines#code-style-guidelines)
- [ ] Tests have been added for the bug fix or new feature
- [ ] Docs have been added for the bug fix or new feature

jku reviewed (approved):
The process of building strings just to then split them with shlex seems a little unusual... but I can see how some of the option handling would need more refactoring if you wanted to fix that. LGTM.

lukpueh commented:
> The process of building strings just to then split them with shlex seems a little unusual... but I can see how some of the option handling would need more refactoring if you wanted to fix that.

Exactly. I'd have to refactor how `keyarg` and `homearg` are passed, e.g.:
https://github.com/secure-systems-lab/securesystemslib/blob/6199f462282ea65df06b2e37d003cccdbebd5dba/securesystemslib/gpg/constants.py#L87 

Also, I wasn't sure how we want to treat gpg commands specified in an environment variable. Maybe they include some custom flag, which would behave differently if we stopped using shlex. But maybe that's okay. 🤷  

I mentioned in the commit message that I'm just trying to replicate the original behavior. I can also add a comment in the code.

jku commented:
> I can also add a comment in the code.

Not strictly needed IMO, I think the option and env var handling issues should be pretty obvious to anyone who does try to improve this

lukpueh commented:
- Rebased w/o change on https://github.com/secure-systems-lab/securesystemslib/pull/504
- Fixed linter issues
- Applied Jussi's suggestion to use `capture_output` instead of stdout/stderr PIPEs

CI passes. Merging...
